MemorialHouse/MHPresentation/MHPresentation/Source/Category/CategoryViewController.swift

```diff
@@ -73,22 +89,45 @@ final class CategoryViewController: UIViewController {
         ]
         
         // 좌측 편집 버튼
-        navigationItem.leftBarButtonItem = UIBarButtonItem(
+        let editButton = UIBarButtonItem(
             title: "편집",
             normal: normalAttributes,
             selected: selectedAttributes
         ) { [weak self] in
-            // TODO: 편집하기
-            self?.categoryTableView.setEditing(true, animated: true)
+            guard let self else { return }
+            let isEditing = self.categoryTableView.isEditing
+            self.categoryTableView.setEditing(!isEditing, animated: true)
+            
+            // 버튼 타이틀 업데이트
+            let newTitle = isEditing ? "편집" : "완료"
+            self.navigationItem.leftBarButtonItem?.title = newTitle
         }
+        navigationItem.leftBarButtonItem = editButton
         
         // 우측 추가 버튼
         navigationItem.rightBarButtonItem = UIBarButtonItem(
             title: "추가",
             normal: normalAttributes,
             selected: selectedAttributes
         ) { [weak self] in
-            // TODO: 로컬에 저장하고 테이블뷰에 추가하는 로직 필요
+            guard let self else { return }
+            
+            let alert = UIAlertController(
+                title: "카테고리 추가",
+                message: "새로운 카테고리를 입력해주세요.",
+                textFieldConfiguration: { textField in
+                    textField.placeholder = "카테고리 이름"
+                },
+                confirmHandler: { [weak self] newText in
+                    guard let newText = newText, !newText.isEmpty else {
```

Comment:
P3:
nil값 체크라면 self 옵셔널 체크와 동일하게 그냥 newText만 써주셔도 됩니당 ~
```suggestion
                    guard let newText, !newText.isEmpty else {
```